drivers/regulator/regulator_axp2101.c

```diff
@@ -0,0 +1,557 @@
+{
+	const struct regulator_axp2101_config *config = dev->config;
+
+	*curr_ua = config->desc->max_ua;
+
+	return 0;
+}
+
+static struct regulator_driver_api api = {
+	.enable = axp2101_enable,
+	.disable = axp2101_disable,
+	.count_voltages = axp2101_count_voltages,
+	.list_voltage = axp2101_list_voltage,
+	.set_voltage = axp2101_set_voltage,
+	.get_voltage = axp2101_get_voltage,
+	.set_mode = axp2101_set_mode,
+	.get_current_limit = axp2101_get_current_limit,
+};
+
+static int regulator_axp2101_init(const struct device *dev)
+{
+	const struct regulator_axp2101_config *config = dev->config;
+	uint8_t enabled_val;
+	bool is_enabled;
+	int ret = 0;
+
+	regulator_common_data_init(dev);
+
+	/* read regulator state */
+	ret = i2c_reg_read_byte_dt(&config->i2c, config->desc->enable_reg, &enabled_val);
+	if (ret != 0) {
+		LOG_INST_ERR(config->log, "Reading enable status failed!");
+		return ret;
+	}
+	is_enabled = ((enabled_val & config->desc->enable_mask) == config->desc->enable_val);
+	LOG_INST_DBG(config->log, "is_enabled: %d", is_enabled);
+
+	return regulator_common_init(dev, is_enabled);
+}
+
+#define REGULATOR_AXP2101_DEFINE(node_id, id, name)                                                 \
+	static struct regulator_axp2101_data data_##id;                                             \
+	LOG_INSTANCE_REGISTER(name, node_id, CONFIG_REGULATOR_LOG_LEVEL);                          \
+	static const struct regulator_axp2101_config config_##id = {                                \
+		.common = REGULATOR_DT_COMMON_CONFIG_INIT(node_id),                                \
+		.desc = &name##_desc,                                                              \
+		.i2c = I2C_DT_SPEC_GET(DT_GPARENT(node_id)),                                       \
+		LOG_INSTANCE_PTR_INIT(log, name, node_id)};                                        \
+	DEVICE_DT_DEFINE(node_id, regulator_axp2101_init, NULL, &data_##id, &config_##id,           \
+			 POST_KERNEL, CONFIG_REGULATOR_AXP2101_INIT_PRIORITY, &api);
+
+#define REGULATOR_AXP2101_DEFINE_COND(inst, child)                                                  \
+	COND_CODE_1(DT_NODE_EXISTS(DT_INST_CHILD(inst, child)),                                    \
+		    (REGULATOR_AXP2101_DEFINE(DT_INST_CHILD(inst, child), child##inst, child)), ())
+
+#define REGULATOR_AXP2101_DEFINE_ALL(inst)                                                       \
+	REGULATOR_AXP2101_DEFINE_COND(inst, dcdc1)                                                   \
+	REGULATOR_AXP2101_DEFINE_COND(inst, dcdc2)                                                   \
+	REGULATOR_AXP2101_DEFINE_COND(inst, dcdc3)                                                   \
+    REGULATOR_AXP2101_DEFINE_COND(inst, dcdc4)                                                   \
+    REGULATOR_AXP2101_DEFINE_COND(inst, dcdc5)                                                   \
+	REGULATOR_AXP2101_DEFINE_COND(inst, ldoa1)                                                   \
```

Comment:
Please align the name to the datasheet notation.
Here, ldoa1-> aldo1.
The others so on.

Reply:
It is also need to fix dts and commit messages.